tests/CMakeLists.txt

```
@@ -5,6 +5,24 @@ foreach( _conf ${CMAKE_CONFIGURATION_TYPES} )
   set( CMAKE_ARCHIVE_OUTPUT_DIRECTORY_${_conf} ${CMAKE_CURRENT_BINARY_DIR} )
 endforeach()
 
+# we need to boostrap the data model, so this has to be executed in the cmake run
+execute_process(
+  COMMAND ${CMAKE_COMMAND} -E echo "Creating TestDataModel"
+  COMMAND python ${CMAKE_SOURCE_DIR}/python/podio_class_generator.py tests/datalayout.yaml tests datamodel
+  WORKING_DIRECTORY ${CMAKE_SOURCE_DIR})
+
+# This ensures that we re-run the class generator if the yaml file is touched
+# the command is run if datalayout.yaml is newer then selection.xml
+add_custom_command(OUTPUT ${CMAKE_SOURCE_DIR}/tests/src/selection.xml  # we specify one output file so this doesn't always run
+  COMMENT "Re-Creating TestDataModel"
+  MAIN_DEPENDENCY datalayout.yaml
+  # This removal ensures selection.xml is newer than datalayout.yaml, other wise this command always runs
+  COMMAND ${CMAKE_COMMAND} -E remove ${CMAKE_SOURCE_DIR}/tests/src/selection.xml
```

While this re-runs the class generator when the yaml file changes, it comes to late in the process.
Changing (bringing the GLOB lines above)
```suggestion
  COMMAND ${CMAKE_COMMAND} -E remove ${CMAKE_SOURCE_DIR}/tests/src/selection.xml ${sources} ${headers}
```
Will make `make` recompile everything because the object files would be older than the yaml file. `ninja` is better in this regard.

---

Great! Looking forward to testing it, once the WIP flag is gone

---

No, it is not great at all :) My goal was to only re-compile what is needed...